knightbus-postgresql/src/KnightBus.PostgreSql/PostgresBus.cs

```diff
@@ -83,4 +96,19 @@ public PostgresBus(NpgsqlDataSource npgsqlDataSource, IPostgresConfiguration pos
         await command.PrepareAsync(ct);
         await command.ExecuteNonQueryAsync(ct);
     }
+    private async Task PublishAsyncInternal<T>(IEnumerable<T> messages, CancellationToken ct) where T : IPostgresEvent
+    {
+        var topicName = AutoMessageMapper.GetQueueName<T>();
+        await using var connection = await _npgsqlDataSource.OpenConnectionAsync(ct);
+
+        await using var cmd = new NpgsqlCommand($"select {SchemaName}.publish_events(@topic, @messages)", connection);
+
+        var serialized = messages.Select(m => _serializer.Serialize(m)).ToArray();
+
+        cmd.CommandType = CommandType.Text;
+        cmd.Parameters.Add(new NpgsqlParameter { ParameterName = "topic", Value = topicName, NpgsqlDbType = NpgsqlDbType.Text });
+        cmd.Parameters.Add(new NpgsqlParameter { ParameterName = "messages", Value = serialized, NpgsqlDbType = NpgsqlDbType.Array | NpgsqlDbType.Jsonb });
```

Comment:
Use positional parameters ($1, $2)  and get rid of ParameterName so npgsql can go into “raw mode”, where it does not perform any parsing or rewriting of your sql.

```suggestion
        await using var cmd = new NpgsqlCommand($"select {SchemaName}.publish_events($1, $2)", connection);

        var serialized = messages.Select(m => _serializer.Serialize(m)).ToArray();
        
        cmd.CommandType = CommandType.Text;
        cmd.Parameters.Add(new NpgsqlParameter { Value = topicTableName, NpgsqlDbType = NpgsqlDbType.Text});
        cmd.Parameters.Add(new NpgsqlParameter { Value = serialized, NpgsqlDbType = NpgsqlDbType.Array | NpgsqlDbType.Jsonb});

```